staging/src/k8s.io/apiserver/pkg/util/webhook/webhook.go

```diff
@@ -90,6 +92,15 @@ func newGenericWebhook(scheme *runtime.Scheme, codecFactory serializer.CodecFact
 
 	codec := codecFactory.LegacyCodec(groupVersions...)
 	clientConfig.ContentConfig.NegotiatedSerializer = serializer.NegotiatedSerializerWrapper(runtime.SerializerInfo{Serializer: codec})
+	if egressLookup != nil {
+		networkContext := egressselector.Master.AsNetworkContext()
```

Comment:
is this context valid for every webhook type? authentication, audit, authorization, imagepolicy setup all use this path

Reply:
I would have expected to pass egressDialer in, rather than do the lookup here

Reply:
Yes, all webhooks (with the exception of admission webhooks that don't go through this code path) can share the same context. This PR only sets it up for authentication, but will be extended to audit, authorization, and imagepolicy soon.

Reply:
My point is that this makes assumptions that don't hold for generic webhooks inside a method named "newGenericWebhook". We should either accept a dialer or network context as a param, or rename the method to make it clear that only webhooks that expect to run in the master network context should call this.

Reply:
Good point. Refactored.